relay-server/src/actors/project.rs

```diff
@@ -379,6 +382,40 @@ enum GetOrFetch<'a> {
     Scheduled(&'a mut StateChannel),
 }
 
+/// Enum representing either the project state or an aggregation of metrics.
+///
+/// We want to wait with enforcing rate limits on metrics until we have a project state,
+/// So when we don't have one yet, we hold them in this aggregator until the project state arrives.
+#[derive(Debug)]
+#[allow(clippy::large_enum_variant)]
+enum AggregationState {
+    State(Arc<ProjectState>),
+    Aggregator(aggregator::Aggregator),
```

Comment:
IMO it's better to describe the state itself instead of the data that are contained in each variant.

```suggestion
    Cached(Arc<ProjectState>),
    Pending(aggregator::Aggregator),
```